app/bin/tools/datastore.dart

```diff
@@ -0,0 +1,335 @@
+        .replaceAll(':', '')
+        .split('.')
+        .first;
+    // output files
+    final packagesFile = new File('${backupDir.path}/$ts-packages.jsonl');
+
+    await withProdServices(() async {
+      print('Backup packages into ${packagesFile.path}...');
+      int pkgCounter = 0;
+      await for (final p in dbService.query<Package>().run()) {
+        pkgCounter++;
+        if (pkgCounter == 1 || pkgCounter % 20 == 0) {
+          print('Processing #$pkgCounter [${p.name}]...');
+        }
+        final versions = await dbService
+            .query<PackageVersion>(ancestorKey: p.key)
+            .run()
+            .toList();
+
+        final archiveLine = new ArchiveLine(
+          package: new PackageArchive(
+            name: p.name,
+            created: p.created,
+            updated: p.updated,
+            downloads: p.downloads,
+            latestVersion: p.latestVersion,
+            latestDevVersion: p.latestDevVersion,
+            uploaderEmails: p.uploaderEmails,
+            isDiscontinued: p.isDiscontinued,
+            doNotAdvertise: p.doNotAdvertise,
+          ),
+          versions: versions
+              .map((pv) => new PackageVersionArchive(
+                    version: pv.version,
+                    created: pv.created,
+                    pubspecJson: convert.json.encode(pv.pubspec.asJson),
+                    readmeFilename: pv.readmeFilename,
+                    readmeContent: pv.readmeContent,
+                    changelogFilename: pv.changelogFilename,
+                    changelogContent: pv.changelogContent,
+                    exampleFilename: pv.exampleFilename,
+                    exampleContent: pv.exampleContent,
+                    libraries: pv.libraries,
+                    downloads: pv.downloads,
+                    sortOrder: pv.sortOrder,
+                    uploaderEmail: pv.uploaderEmail,
+                  ))
+              .toList(),
+        );
+
+        final line = convert.json.encode(archiveLine.toJson());
+        await packagesFile.writeAsString(line + '\n',
+            mode: FileMode.writeOnlyAppend);
+      }
+      print('$pkgCounter packages backed up in ${packagesFile.path}');
+    });
+  }
+}
+
+/// 'Updates user-provided data from backup.'
+class UpdateCommand extends Command {
+  @override
+  String get name => 'update';
+
+  @override
+  String get description => 'Updates user-provided data from backup.';
+
+  UpdateCommand() {
+    argParser.addOption('backup-directory', help: 'The backup directory.');
+  }
+
+  @override
+  Future run() async {
+    if (activeConfiguration.projectId != 'dartlang-pub-dev') {
+      print('\nPROJECT_ID is not dartlang-pub-dev, exiting.\n');
+      exit(-1);
+    }
+
+    final backupDir = argResults['backup-directory'] == null
+        ? Directory.current
+        : new Directory(argResults['backup-directory'] as String);
+    // input files
+    final backupPrefix = await _detectLatest(backupDir);
```

Comment:
It seems much simpler and more flexible to just have the user provide the correct backup path on the command line?

Reply:
I thought that a cron job would be easier if it doesn't need to specify the exact file name. Maybe have it as an optional argument?

Reply:
In a cron job, it's more useful if the output mechanism lets me supply the output filename, and then it can provide it to the restore step.

Reply:
PTAL, I've updated both commands to use explicit files.